tap_postgres/tap.py

```diff
@@ -33,7 +80,82 @@ def connector(self) -> PostgresConnector:
         Connector is a singleton (one instance is used by the Tap and Streams).
 
         """
-        return PostgresConnector(dict(self.config))
+        # We mutate this url to use the ssh tunnel if enabled
+        self.url = make_url(self.config["sqlalchemy_url"])
+
+        if self.config["ssh_tunnel.enable"]:
+            self.ssh_tunnel_connect()
+        return PostgresConnector(
+            config=dict(self.config),
+            sqlalchemy_url=str(
+                self.url
+            ),  # Sometimes we mutuate the url to use the ssh tunnel
+        )
```

Comment:
Ditto my other comment about removing side effects:

```suggestion
        # We mutate this url to use the ssh tunnel if enabled
        url = make_url(self.config["sqlalchemy_url"])

        if self.config["ssh_tunnel.enable"]:
            # Return a new URL with SSH tunnel parameters
            url = self.ssh_tunnel_connect(url)
        return PostgresConnector(
            config=dict(self.config),
            sqlalchemy_url=str(self.url)
        )
```